### PR TITLE
Add RcParam for HTTP protocol for remote dataset fetching

### PR DESCRIPTION
**Pull request by canyon289**

### CODE REVIEW

**Comment by OriolAbril** [comment]
How about using a rcparam `data.allow_http` or `data.use_https` so that the default keeps being using https? 

It could be handled when loading instead of at url definition (I think a `str.replace("https", "http")` should do)

**Comment by canyon289** [comment]
I'm good with that! I'll update PR later this week

File: arviz/data/datasets.py [hunk 1]
```diff
@@ -213,8 +215,14 @@ def load_arviz_data(dataset=None, data_home=None):
         remote = REMOTE_DATASETS[dataset]
         home_dir = get_data_home(data_home=data_home)
         file_path = os.path.join(home_dir, remote.filename)
+
         if not os.path.exists(file_path):
-            urlretrieve(remote.url, file_path)
+            http_type = RcParams["data.http_protocol"]
+
+            # Replaces http type. Redundant if http_type is http, useful if http_type is https
+            url = remote.url.replace("http", http_type)
```
Comment (canyon289) [inline arviz/data/datasets.py]: @OriolAbril Let me know if this string replace method is fine by you. I can always do the string substitution method as well. Picked current pattern so urls would be valid if anyone wanted to copy and paste but I'm not tied to the pattern

File: arviz/data/datasets.py [hunk 2]
```diff
@@ -7,6 +7,8 @@
 from urllib.request import urlretrieve
 
 from .io_netcdf import from_netcdf
+from ..rcparams import RcParams
```
Comment (OriolAbril) [inline arviz/data/datasets.py]: RcParams i camel case is the name of the class defined, the specific instance to customize arviz is `rcParams` (using the same names as matplotlib which I figured would be less confusing)

File: arvizrc.template [hunk 3]
```diff
@@ -1,6 +1,8 @@
 #### ArviZ rcParams template ####
 ### DATA  ###
 # rcParams related with data loading related functions
+data.http_protocol           : http  # protocol for remote dataset load,
```
Comment (OriolAbril) [inline arvizrc.template]: This should be https, to match the default in rcparams.py

**Pull request merged**